controllers/automation/packagedeployment_controller.go

```diff
@@ -230,11 +236,16 @@ metadata:
 	// If a binding resource exists in the package, and one exists
 	// that matches the name of the associated cluster, overwrite the
 	// resource in the package with the cluster resource
+	conditions := convertConditions(newPR.Status.Conditions)
 
 	for id, pkgResource := range bindingResources {
 		clusterResource, err := r.findClusterObject(ctx, c, id)
+		group, _ := resid.ParseGroupVersion(id.APIVersion)
+		conditionType := fmt.Sprintf("%s.%s.%s.%s.Injected", group, id.Kind, c.Name, c.Namespace)
```

Comment:
should we lowercase the condition type? is that typical? (maybe see if there is anything in k8s API conventions about that)

Reply:
Looking at pods for example, the condition types seem to be camelcase like `PodScheduled` or `ContainersReady`. Here I was trying to have a unique condition type for each injected resource, so using
the resourceID seemed like a natural way to do it. In my original PR, it looked something like `infra.nephio.org.ClusterScaleProfile.nginx.default.Injected`.

But if we want it to look more like the condition types for builtin resources, maybe we can leave out the group and try to camelcase things, so something like `ClusterScaleProfile.Nginx.Default.Injected`. My most recent commit makes this change, but please let me know if you have some other format ideas.

Reply:
1) Can we have more than one condition of the same type, or is type the identifier?
2) I think we need to keep the group, even if it's not as pretty, for the reason you said - uniqueness.

Assuming the answer to 1) is that there is only one condition of the same type, then your original is the right way. We will for example, have many different `ipam.nephio.org.IPAMAllocation.*.*.Injected` conditions in a single package.

Reply:
Yeah, the apimachinery libraries treat `Type` as the identifier. I'll revert it back to what I did before. So we'll get something like 

```
status:
  conditions:
  - message: resource with apiVersion "infra.nephio.org/v1alpha1", kind "ClusterContext",
      name "edge-1", and namespace "default" injected to the package revision from
      the cluster
    reason: ResourceInjectedFromCluster
    status: "True"
    type: infra.nephio.org.ClusterContext.edge-1.default.Injected
  - message: upfdeployments.nf.nephio.org "edge-1" not found
    reason: ResourceNotFound
    status: "False"
    type: nf.nephio.org.UPFDeployment.edge-1.default.Injected
```

Reply:
nice!